utils/get_test_info.py

```diff
@@ -0,0 +1,118 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import importlib
+import os
+import sys
+import unittest
+
+
+# This is required to make the module import works (when the python process is running from the root of the repo)
+sys.path.append(".")
```

Comment:
This is the missing piece you needed (I believe/guess) in `check_repo.py` before, when you wrote

```python
# This is a bit hacky but I didn't find a way to import the test_file as a module and read inside the class
```